java-squid/src/main/java/org/sonar/java/JavaSquid.java

```diff
@@ -76,8 +79,11 @@ public JavaSquid(JavaConfiguration conf,
       Iterable<CodeVisitor> measurers = Arrays.asList((CodeVisitor)measurer);
       visitorsToBridge =  Iterables.concat(visitorsToBridge, measurers);
     }
-
-    VisitorsBridge visitorsBridge = new VisitorsBridge(visitorsToBridge, sonarComponents);
+    List<File> classpath = Lists.newArrayList();
+    if(sonarComponents != null) {
```

Comment:
The test `sonarComponents != null` is repeated 4 times in a row. It can be simplified.